lib/rstuf/client.rb

```diff
@@ -0,0 +1,37 @@
+class Rstuf::Client
+  Error = Class.new(StandardError)
+
+  def self.post_artifacts(targets)
+    response = connection.post("/api/v1/artifacts/", { targets: targets })
+
+    return response.body.dig("data", "task_id") if response.success?
+    raise Error, "Error posting artifacts: #{response.body}"
+  end
+
+  def self.delete_artifacts(targets)
+    response = connection.post("/api/v1/artifacts/delete", { targets: targets }, {})
+
+    return response.body.dig("data", "task_id") if response.success?
+    raise Error, "Error deleting artifacts: #{response.body}"
+  end
+
+  def self.task_status(task_id)
```

Comment:
I suggest changing it to `task_state` as it retrieves the task `state`, which contains the task `status`.
```suggestion
  def self.task_state(task_id)
```